### ISSUE TITLE
[BUG] static resources changed from relative to absolute urls in 3.3.0

### ISSUE THREAD

**Issue by jacobtomlinson** [bokeh/bokeh#13521]
### Software versions

Python version      :  3.10.11 | packaged by conda-forge | (main, May 10 2023, 19:07:22) [Clang 14.0.6 ]
IPython version     :  8.13.2
Tornado version     :  6.3.3
Bokeh version       :  3.3.0
BokehJS static path :  /Users/jtomlinson/miniconda3/envs/dask/lib/python3.10/site-packages/bokeh/server/static
node.js version     :  (not installed)
npm version         :  (not installed)
Operating system    :  macOS-13.5.2-x86_64-i386-64bit

### Expected behavior

In `bokeh==3.2.2` static urls were relative.

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    ...
    <script type="text/javascript" src="static/js/bokeh.min.js?v=3ca6425586de5036dc01992dd69aa61e9196dd02619557cfaeb1b3d8b77adf724be49401b1168483d165494ce57a6daa16e6f6d3660fef117d45028221f86357"></script>
...
```

### Observed behavior

However after upgrading to `3.3.0` they became absolute.

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    ...
    <script type="text/javascript" src="/static/js/bokeh.min.js?v=39ef57c3a83533e24f961e5c27f651a61045dbccefac4b5df86a7680b1edaff31886a7c0322250ffb0d758fa14ae156c9b640f60cca99f020096b050a4dbb571"></script>
...
```

This causes problems when we try to view the page via some proxy outside of our control that puts our server on a different path. Absolute URLs cause the JS and CSS files to be looked for in the wrong place.

### Example code

In Dask we already have a tornado server so we use the `BokehTornado` server to add it. This is a distilled version of what we do in Dask that reproduces the problem.

```python
from bokeh.server.server import BokehTornado
from tornado.httpserver import HTTPServer
from tornado.ioloop import IOLoop

bokeh_app = BokehTornado({"/": lambda _: None})
http_server = HTTPServer(bokeh_app)
http_server.listen(port=1234)
IOLoop.current().start()
```

Then visit http://localhost:1234 and view the page source.

**Comment by bryevdv** [comment]
Offhand I might suspect https://github.com/bokeh/bokeh/pull/13391 which was in response to another regression from  https://github.com/bokeh/bokeh/pull/13041.

**Comment by bryevdv** [comment]
I believe this code exhibits the desired behavior (waiting on confirm from @jacobtomlinson)
```python
from bokeh.resources import Resources
from bokeh.server.server import BokehTornado
from tornado.httpserver import HTTPServer
from tornado.ioloop import IOLoop

class DaskBokehTornado(BokehTornado):
    def resources(self, absolute_url=True):  # new default True
        return super().resources(absolute_url)

bokeh_app = DaskBokehTornado({"/": lambda _: None}, absolute_url="")
http_server = HTTPServer(bokeh_app)
http_server.listen(port=1234)
IOLoop.current().start()
```

If that is the case, then the task is to decide what is the best course of action for Bokeh? Comments/questions:

* Change default value of `absolute_url` in `resources` to True? That would allow Dask to configure `absolute_url=""` value for `BokehTornado` which is bit weird but if it works...
* Not sure what the wider implications of that would change would be. FWIW I am not actually sure anywhere in the codebase calls `resources` with anything other then default args, so maybe this can/should be simplified in general. 
* Backtrack both the PRs mentioned above, and address the original inciting issue about port configuration in a different way? 
* Something else?

Regardless, one thing I was very surprised to remember that it's not possible to just manually provide a `Resources` for `BokehTornado` to use unconditionally. I am sure there were historical reasons that made sense, but I think nowadays we just need to expose that option. That would allow e.g. Dask to just be explicit and insulate themselves from relying on default behavior.

**Comment by bryevdv** [comment]
I guess its also worth checking for good measure for Dask specifically: @jacobtomlinson are CDN resources an option? Using CDN resources would sidestep all of this entirely, and bring some other benefits as well [1], but of course they would not be an option for anyone in an airgapped or otherwise isolated network that could not access `cdn.bokeh.org`

[1] browsers could actually cache BokehJS, the load on the server would be reduced

**Comment by jacobtomlinson** [comment]
Thanks @bryevdv. I'll look at adding this shim into Dask to unblock things now (https://github.com/dask/distributed/pull/8347).

> are CDN resources an option?

Ideally we would like the Dask Dashboard to work in environments without an internet connection, so relying on CDN resources isn't ideal.

**Comment by jacobtomlinson** [comment]
Just a quick update here, the workaround has been merged into Dask so we aren't waiting on this.
